pkg/spoke/controllers/deletioncontroller/deletion_controller.go

```diff
@@ -96,6 +102,15 @@ func (m *StaleManifestDeletionController) syncManifestWork(ctx context.Context,
 		})
 	}
 
+	// delete applied resources which are no longer maintained by manifest work
+	noLongerMaintainedResources := findUntrackedResources(manifestWork.Status.AppliedResources, appliedResources)
+	resourcesPendingFinalization, errs := helper.DeleteAppliedResources(noLongerMaintainedResources, m.spokeDynamicClient)
+	if len(errs) != 0 {
+		return utilerrors.NewAggregate(errs)
+	}
+
+	appliedResources = append(appliedResources, resourcesPendingFinalization...)
```

Comment:
minor: you need to de-dup these lists.

Reply:
`appliedResources`, the second one on the above line, are built based on ManifestConditions in status. It contains the resources currently maintained by the manifest work. `resourcesPendingFinalization` is a subset of resources which are no longer maintained by manifest work. So I think there is no duplicate item in the concated slice.